.github/workflows/build.yaml

```diff
@@ -32,39 +32,32 @@ jobs:
           - 6379:6379
 
     steps:
-      - uses: actions/checkout@v1
-      - uses: actions/setup-node@v1
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v3
+        with:
+          cache: yarn
       - uses: ruby/setup-ruby@v1
         with:
           bundler-cache: true # runs 'bundle install' and caches installed gems automatically
       - name: Install PostgreSQL 11 client
         run: sudo apt-get update && sudo apt-get install postgresql postgresql-contrib libpq-dev --fix-missing
-      - name: Get yarn cache
-        id: yarn-cache
-        run: echo "::set-output name=dir::$(yarn cache dir)"
-      - uses: actions/cache@v1
-        with:
-          path: ${{ steps.yarn-cache.outputs.dir }}
-          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
-          restore-keys: |
-            ${{ runner.os }}-yarn-
       - name: Install yarn && dependencies
-        run: yarn install --ignore-engines
-
+        run: yarn install
       - name: Prepare database
         env:
           PGHOST: localhost
           PGPASSWORD: password
           RAILS_ENV: test
-        run: |
-          bundle exec rake db:test:prepare
-
+        run: bundle exec rake db:test:prepare
+      - name: Build assets
+        env:
+          RAILS_ENV: test
+        run: NODE_OPTIONS=--openssl-legacy-provider bundle exec rake assets:precompile
```

Comment:
This is kinda funky. I always had to build the assets (`/public/packs` and `/public/packs-test`) locally in order to get the tests run. It did work in CI however. I figure that these folders where somehow cached and that is why there was some (probably outdated) javascript used that made the tests pass. When I switched to the newer packages and auto-caching, the tests started failing, so I added the step also in CI. I created #703 to show why this step is important.